app/core/authentication.js

```diff
@@ -0,0 +1,43 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+import FauxtonAPI from "./base";
+
+export const authenticate = (session, roles) => {
+  if (session.isAdminParty()) {
+    return true;
+  } else if (session.matchesRoles(roles)) {
```

Comment:
Normally I would agree with you. But I would prefer to keep them separate since they involve different checks. I feel this makes it a little more obvious.